lib/astarte_data_access/xandra_utils.ex

```diff
@@ -23,67 +23,120 @@ defmodule Astarte.DataAccess.XandraUtils do
 
   require Logger
 
-  @spec run(String.t(), (Xandra.Connection.t(), String.t() -> any)) ::
-          any | {:error, :invalid_realm_name}
-  def run(realm, fun) do
-    with :ok <- verify_realm(realm) do
-      keyspace = CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
+  @cluster_default_name :astarte_data_access_xandra
 
-      Xandra.Cluster.run(:astarte_data_access_xandra, &fun.(&1, keyspace))
+  def validate_realm_name(realm_name) do
+    if Realm.valid_name?(realm_name) do
+      :ok
+    else
+      Logger.warning("Invalid realm name.",
+        tag: "invalid_realm_name",
+        realm: realm_name
+      )
```

Comment:
No need to log this, as the error will be handled by the context around the caller